test/integration-tests/Uniswap.test.ts

```diff
@@ -1196,6 +1205,436 @@ describe('Uniswap V2 and V3 Tests:', () => {
+
+        planner.addCommand(CommandType.V3_POSM_CALL, [encodedCall])
+        await executeRouter(planner)
+
+        position = await v3NFTPositionManager.positions(tokenId)
+        liquidity = position.liquidity
+
+        expect(liquidity).to.eq(0)
+      })
+      it('cannot call decrease liquidity with improper signature', async () => {
+        // first we need to permit the router to spend the nft
+        const { v, r, s } = await getPermitNFTSignature(bob, v3NFTPositionManager, router.address, tokenId, MAX_UINT)
+        planner.addCommand(CommandType.ERC721_PERMIT, [router.address, tokenId, MAX_UINT, v, r, s])
+
+        let position = await v3NFTPositionManager.positions(tokenId)
+        let liquidity = position.liquidity
+
+        const BAD_DECREASE_LIQUIDITY_STRUCT =
+          '(uint256 tokenId,uint256 liquidity,uint256 amount0Min,uint256 amount1Min,uint256 deadline)'
+
+        const params = { tokenId: tokenId, liquidity: liquidity, amount0Min: 0, amount1Min: 0, deadline: MAX_UINT }
+
+        const abi = new ethers.utils.AbiCoder()
+        const encodedParams = abi.encode([BAD_DECREASE_LIQUIDITY_STRUCT], [params])
+        const functionSignature = ethers.utils
+          .id('decreaseLiquidity((uint256,uint128,uint256,uint256))')
+          .substring(0, 10)
+        const encodedCall = functionSignature + encodedParams.substring(2)
+
+        planner.addCommand(CommandType.V3_POSM_CALL, [encodedCall])
+        await expect(executeRouter(planner)).to.be.revertedWithCustomError(router, 'InvalidV3Action')
+      })
+      it('fails if decrease liquidity call fails', async () => {
+        // first we need to permit the router to spend the nft
+        const { v, r, s } = await getPermitNFTSignature(bob, v3NFTPositionManager, router.address, tokenId, MAX_UINT)
+        planner.addCommand(CommandType.ERC721_PERMIT, [router.address, tokenId, MAX_UINT, v, r, s])
+
+        let position = await v3NFTPositionManager.positions(tokenId)
+        let liquidity = position.liquidity
+
+        const DECREASE_LIQUIDITY_STRUCT =
+          '(uint256 tokenId,uint256 liquidity,uint256 amount0Min,uint256 amount1Min, uint256 deadline)'
+
+        const params = { tokenId: tokenId, liquidity: liquidity, amount0Min: 0, amount1Min: 0, deadline: 0 }
+
+        const abi = new ethers.utils.AbiCoder()
+        const encodedParams = abi.encode([DECREASE_LIQUIDITY_STRUCT], [params])
+        const functionSignature = ethers.utils
+          .id('decreaseLiquidity((uint256,uint128,uint256,uint256,uint256))')
+          .substring(0, 10)
+        const encodedCall = functionSignature + encodedParams.substring(2)
+
+        planner.addCommand(CommandType.V3_POSM_CALL, [encodedCall])
+
+        await expect(executeRouter(planner)).to.be.revertedWithCustomError(router, 'CallToV3PositionManagerFailed')
+      })
+      it('cannot call decrease liquidity if not approved', async () => {
```

Comment:
would also love a test where no `ERC721_PERMIT` command is provided to show that fails too